zerver/tests/test_retention.py

```diff
@@ -26,89 +31,290 @@ def setUp(self):
-            [message.id for message in expired_messages_result[1]['expired_messages']]
+            list(
+                UserMessage.objects
+                    .filter(message__in=exp_zulip_msgs)
+                    .order_by('id')
+                    .values_list('id', flat=True)
+            ),
+            list(
+                ArchiveUserMessage.objects
+                    .filter(message__in=archived_messages, user_profile__realm=self.zulip_realm)
+                    .order_by('id')
+                    .values_list('id', flat=True)
+            )
         )
 
     def test_expired_messages_in_one_realm(self):
         # type: () -> None
         # Check realm with expired messages and messages
         # with one day to expiration data.
-        expired_mit_messages = self._make_mit_messages(5, datetime.now() - timedelta(days=101))
-        actual_mit_messages = self._make_mit_messages(3, datetime.now() - timedelta(days=99))
-        expired_messages_result = list(get_expired_messages())
-        expired_mit_messages_ids = [message.id for message in expired_mit_messages]
-        expired_mit_messages_result_ids = [message.id for message in
-                                           expired_messages_result[0]['expired_messages']]
-        actual_mit_messages_ids = [message.id for message in actual_mit_messages]
-        self.assertEqual(len(expired_messages_result), 1)
-        self.assertEqual(len(expired_messages_result[0]['expired_messages']), 5)
-        self.assertEqual(expired_messages_result[0]['realm_id'], self.mit_realm.id)
-        # Compare expected messages ids with result messages ids.
+        exp_mit_msgs = self._make_mit_msgs(5, timezone.now() - timedelta(days=101))
+        move_expired_messages_to_archive()
+        move_expired_user_messages_to_archive()
+        archived_messages = ArchiveMessage.objects.all()
+        archived_user_messages = ArchiveUserMessage.objects.all()
+        self.assertEqual(archived_messages.count(), 5)
+        self.assertEqual(archived_user_messages.count(), 10)
+        # Compare expected messages ids with archived messages in mit realm
+        self.assertEqual(
+            list(exp_mit_msgs.order_by('id').values_list('id', flat=True)),
+            list(
+                archived_messages
+                    .filter(archiveusermessage__user_profile__realm=self.mit_realm)
+                    .order_by('id')
+                    .distinct('id')
+                    .values_list('id', flat=True)
+            )
+        )
+        self.assertEqual(
+            list(
+                UserMessage.objects
+                    .filter(message__in=exp_mit_msgs)
+                    .order_by('id')
+                    .values_list('id', flat=True)
+            ),
+            list(
+                ArchiveUserMessage.objects
+                    .filter(user_profile__realm=self.mit_realm)
+                    .order_by('id')
+                    .values_list('id', flat=True)
```

Comment:
These tests are kinda long; might be worth writing some more test helper functions to simplify repetitive stuff.  

I feel like these tests use the pattern `list(foo.order_by('id').values_list('id', flat=True))` where `foo` is a query that maybe we should make the API here be e.g. `sorted_ids(ArchiveUserMessage.objects                    .filter(user_profile__realm=self.mit_realm))`